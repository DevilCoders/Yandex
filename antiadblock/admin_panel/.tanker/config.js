module.exports = function(config) {
    config.paths.root = './app/i18n';

    config.tanker.force = true;
    config.tanker.project = 'antiadb';
    config.tanker.host = 'tanker-api.yandex-team.ru';
};
