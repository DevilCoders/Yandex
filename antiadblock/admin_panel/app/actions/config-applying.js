import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
