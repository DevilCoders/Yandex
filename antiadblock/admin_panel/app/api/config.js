import {antiadbRequest} from '../lib/api';

export default {
    fetchConfig: function (configId, status, expId) {
        return antiadbRequest({
            path: `/config/${configId}`,
            options: {
                method: 'GET'
            },
            queryParams: {
                status,
                exp_id: expId
            }
        });
    },

    fetchExpConfigsId: function (labelId) {
        return antiadbRequest({
            path: `/label/${labelId}/experiment`,
            options: {
                method: 'GET'
            }
        });
    },

    saveConfig: function (labelId, parentId, data, dataSettings, comment) {
        return antiadbRequest({
            path: `/label/${labelId}/config`,
            options: {
                method: 'POST',
                body: {
                    data: data,
                    data_settings: dataSettings,
                    parent_id: parentId,
                    comment
                }
            }
        });
    },

    applyConfig: function (labelId, configId, options) {
        return antiadbRequest({
            path: `/label/${labelId}/config/${configId}/${options.target}`,
            options: {
                method: 'PUT',
                body: {
                    old_id: options.oldConfigId ? options.oldConfigId : null
                }
            }
        });
    },

    fetchCurrentConfig: function (labelId, target) {
        return antiadbRequest({
            path: `/label/${labelId}/config/${target}`,
            options: {
                method: 'GET'
            }
        });
    },

    setArchivedConfig: function (configId, archived) {
        return antiadbRequest({
            path: `/config/${configId}`,
            options: {
                method: 'PATCH',
                body: {
                    archived
                }
            }
        });
    },

    setModerateConfig: function (labelId, configId, approved, comment) {
        return antiadbRequest({
            path: `/label/${labelId}/config/${configId}/moderate`,
            options: {
                method: 'PATCH',
                body: {
                    approved,
                    comment
                }
            }
        });
    }
};
