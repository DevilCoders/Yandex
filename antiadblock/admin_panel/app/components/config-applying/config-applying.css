.config-applying
{
    &__copy-link-success
    {
        height: 28px;
        line-height: 28px;
    }

    &__copy-button
    {
        display: inline-block;
        margin-top: 15px;
    }
}
