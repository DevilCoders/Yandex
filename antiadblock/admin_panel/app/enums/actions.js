export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
