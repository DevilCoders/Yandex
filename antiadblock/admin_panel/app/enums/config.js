export const STATUS = {
    ACTIVE: 'active',
    INACTIVE: 'inactive',
    TEST: 'test',
    APPROVED: 'approved',
    DECLINED: 'declined',
    EXPERIMENT: 'experiment'
};

export const STATUS_COLORS = {
    [STATUS.ACTIVE]: 'green',
    [STATUS.INACTIVE]: 'gray',
    [STATUS.TEST]: 'blue',
    [STATUS.APPROVED]: 'purple',
    [STATUS.DECLINED]: 'red'
};
