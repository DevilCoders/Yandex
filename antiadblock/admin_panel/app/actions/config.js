import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

import {antiadbUrl} from 'app/lib/url';
import redirect from 'app/lib/redirect';
import {getStatus} from 'app/lib/config-editor';

export const START_FETCH_CONFIG_LOADING = 'START_FETCH_CONFIG_LOADING';
export const END_FETCH_CONFIG_LOADING = 'END_FETCH_CONFIG_LOADING';
export const START_CONFIG_SAVING = 'START_CONFIG_SAVING';
export const END_CONFIG_SAVING = 'END_CONFIG_SAVING';

function prepareValidation(errors = []) {
    return errors.reduce((validation, error) => {
        switch (error.path[0]) {
            case 'comment':
                validation.comment.push(error);
                break;
            case 'data':
                if (!validation.data[error.path[1]]) {
                    validation.data[error.path[1]] = [];
                }

                validation.data[error.path[1]].push(error);
                break;
            default:
                break;
        }

        return validation;
    }, {
        comment: [],
        data: {}
    });
}

export function fetchConfigById(configId, status, expId) {
    return dispatch => {
        dispatch(startConfigLoading(configId));

        return configApi.fetchConfig(configId, status, expId).then(data => {
            return dispatch(endConfigLoading(configId, data));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function saveConfig(serviceId, labelId, parentId, comment, data, dataSettings) {
    return dispatch => {
        dispatch(startConfigSaving());

        return configApi.saveConfig(labelId, parentId, comment, data, dataSettings).then(() => {
            redirect(antiadbUrl(`/service/${serviceId}/label/${labelId}/`));
        }, error => {
            if (error.status === 400) {
                const properties = error.response.properties;
                const validation = prepareValidation(properties);

                return dispatch(endConfigSaving(validation));
            }
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startConfigSaving() {
    return {
        type: START_CONFIG_SAVING
    };
}

export function endConfigSaving(validation) {
    return {
        type: END_CONFIG_SAVING,
        validation
    };
}

export function startConfigLoading(configId) {
    return {
        type: START_FETCH_CONFIG_LOADING,
        configId
    };
}

function getConfigStatus(data) {
    const statuses = (data.statuses && data.statuses[0] && data.statuses[0].status && data.statuses.map(item => item.status)) || [];

    return getStatus(statuses, data.exp_id);
}

export function endConfigLoading(configId, data) {
    return {
        type: END_FETCH_CONFIG_LOADING,
        configId,
        data: {
            ...data,
            dataCurrent: data.data,
            status: getConfigStatus(data)
        }
    };
}
