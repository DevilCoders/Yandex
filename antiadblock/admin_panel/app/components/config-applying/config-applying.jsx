import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import YAML from 'js-yaml';
import {MonacoDiffEditor} from 'react-monaco-editor';

import Bem from 'app/components/bem/bem';

import Modal from 'lego-on-react/src/components/modal/modal.react';
import Button from 'lego-on-react/src/components/button/button.react';

import ModalClose from 'app/components/modal/__close/modal__close';
import LinkCopy from 'app/components/link-copy/link-copy';

import Preloader from '../preloader/preloader';
import {configType} from '../../types/index';

import {fetchCurrentConfig, closeConfigApplying} from 'app/actions/config-applying';
import {applyConfig} from 'app/actions/service';
import {getConfigApplying} from 'app/reducers/index';

import {KEYS} from 'app/enums/keys';

import i18n from 'app/lib/i18n';

import './config-applying.css';
import 'app/components/icon/_theme/icon_theme_link.css';

class ConfigApplying extends React.Component {
    constructor() {
        super();

        this.onApply = this.onApply.bind(this);
        this.onModalClose = this.onModalClose.bind(this);
        this.onKeyUp = this.onKeyUp.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        if (!this.props.visible && nextProps.visible) {
            this.props.fetchCurrentConfig(nextProps.configData.label_id, nextProps.target);
            document.addEventListener('keyup', this.onKeyUp);
        } else if (this.props.visible && !nextProps.visible) {
            document.removeEventListener('keyup', this.onKeyUp);
        }
    }

    onApply() {
        this.props.applyConfig(this.props.configData.label_id, this.props.id, {
            target: this.props.target,
            oldConfigId: this.props.oldConfigData.id
        });
        this.props.closeConfigApplying(this.props.id);
    }

    onModalClose() {
        this.props.closeConfigApplying(this.props.id);
    }

    getLinkDiffConfigToCopyUrl() {
        const {serviceId, configData, oldConfigData} = this.props;

        return `https://${location.host}/service/${serviceId}/configs/diff/${oldConfigData.id}/${configData.id}`;
    }

    onKeyUp(e) {
        if (e.keyCode === KEYS.ENTER) {
            this.onApply();
        }
    }

    render() {
        return (
            <Modal
                autoclosable
                onDocKeyDown={this.onModalClose}
                onOutsideClick={this.onModalClose}
                visible={this.props.visible}>
                {!this.props.loaded ?
                    <Preloader /> :
                    [
                        <Bem
                            key='title'
                            block='modal'
                            elem='title'>
                            {i18n('service-page', 'applying-title')}
                            &nbsp;
                            <LinkCopy
                                text={i18n('hints', 'link-copied-to-buffer')}
                                to='right'
                                value={this.getLinkDiffConfigToCopyUrl()} />
                        </Bem>,
                        <ModalClose
                            key='close'
                            onClick={this.onModalClose} />,
                        <Bem
                            key='body'
                            block='modal'
                            elem='body'>
                            <MonacoDiffEditor
                                language='yaml'
                                options={{
                                    readOnly: true,
                                    scrollBeyondLastLine: false,
                                    minimap: {
                                        enabled: false
                                    }
                                }}
                                original={YAML.safeDump(this.props.oldConfigData.data, {
                                    lineWidth: Infinity
                                })}
                                value={YAML.safeDump(this.props.configData.data, {
                                    lineWidth: Infinity
                                })} />
                        </Bem>,
                        <Bem
                            key='actions'
                            block='modal'
                            elem='actions'>
                            <Button
                                theme='action'
                                view='default'
                                tone='grey'
                                size='s'
                                mix={{
                                    block: 'modal',
                                    elem: 'action'
                                }}
                                onClick={this.onApply}>
                                {i18n('common', 'apply')}
                            </Button>
                            <Button
                                theme='normal'
                                view='default'
                                tone='grey'
                                size='s'
                                mix={{
                                    block: 'modal',
                                    elem: 'action'
                                }}
                                onClick={this.onModalClose}>
                                {i18n('common', 'cancel')}
                            </Button>
                        </Bem>
                    ]}
            </Modal>
        );
    }
}

ConfigApplying.propTypes = {
    id: PropTypes.number,
    visible: PropTypes.bool,
    serviceId: PropTypes.string,
    target: PropTypes.string,
    configData: configType,
    oldConfigData: configType,
    loaded: PropTypes.bool,
    fetchCurrentConfig: PropTypes.func.isRequired,
    applyConfig: PropTypes.func.isRequired,
    closeConfigApplying: PropTypes.func.isRequired
};

export default connect(state => {
    return {
        id: getConfigApplying(state).id,
        visible: getConfigApplying(state).visible,
        serviceId: getConfigApplying(state).serviceId,
        target: getConfigApplying(state).target,
        configData: getConfigApplying(state).configData,
        oldConfigData: getConfigApplying(state).oldConfigData,
        loaded: getConfigApplying(state).loaded
    };
}, dispatch => {
    return {
        fetchCurrentConfig: (labelId, target) => {
            dispatch(fetchCurrentConfig(labelId, target));
        },
        applyConfig: (labelId, configId, target) => {
            dispatch(applyConfig(labelId, configId, target));
        },
        closeConfigApplying: id => {
            dispatch(closeConfigApplying(id));
        }
    };
})(ConfigApplying);
