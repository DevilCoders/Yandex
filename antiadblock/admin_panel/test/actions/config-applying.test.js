import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import {STATUS} from 'app/enums/config';
import * as configApplyingActions from 'app/actions/config-applying';
import * as configApplyingActionsFixtures from '../fixtures/actions/config-applying';
import {CONFIG_ACTIVE, CONFIG_PREVIEW} from '../fixtures/backend/config';
import * as storeFixtures from '../fixtures/store';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('config-applying actions', () => {
    it('should create OPEN_CONFIG_APPLYING', () => {
        expect(configApplyingActions.openConfigApplying(123, 236, CONFIG_PREVIEW, STATUS.ACTIVE)).toEqual(configApplyingActionsFixtures.OPEN_CONFIG_APPLYING);
    });

    it('should create CLOSE_CONFIG_APPLYING', () => {
        expect(configApplyingActions.closeConfigApplying(236)).toEqual(configApplyingActionsFixtures.CLOSE_CONFIG_APPLYING);
    });

    it('should create START_ACTIVE_CONFIG_LOADING', () => {
        expect(configApplyingActions.startActiveConfigLoading(123, STATUS.ACTIVE)).toEqual(configApplyingActionsFixtures.START_ACTIVE_CONFIG_LOADING);
    });

    it('should create END_ACTIVE_CONFIG_LOADING', () => {
        expect(configApplyingActions.endActiveConfigLoading(123, STATUS.ACTIVE, CONFIG_ACTIVE)).toEqual(configApplyingActionsFixtures.END_ACTIVE_CONFIG_LOADING);
    });

    it('should create START_ACTIVE_CONFIG_LOADING and END_ACTIVE_CONFIG_LOADING', () => {
        const store = mockStore(storeFixtures.CONFIG_APPLYING_OPENED);

        fetch.mockResponseOnce(JSON.stringify(CONFIG_ACTIVE));

        return store.dispatch(configApplyingActions.fetchCurrentConfig(123, STATUS.ACTIVE)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_ACTIVE_CONFIG_LOADING and SET_ERRORS', () => {
        const store = mockStore(storeFixtures.CONFIG_APPLYING_OPENED);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(configApplyingActions.fetchCurrentConfig(123, STATUS.ACTIVE)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_ACTIVE_CONFIG_LOADING and try to fetch ACTIVE config after error', () => {
        const store = mockStore(storeFixtures.CONFIG_APPLYING_OPENED);

        fetch.mockResponseOnce('Not found!', {status: 404});

        return store.dispatch(configApplyingActions.fetchCurrentConfig(123, STATUS.TEST)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_ACTIVE_CONFIG_LOADING and try to fetch ACTIVE config after error', () => {
        const store = mockStore(storeFixtures.CONFIG_APPLYING_OPENED);

        fetch.mockResponseOnce('Not found!', {status: 404});

        return store.dispatch(configApplyingActions.fetchCurrentConfig(123, STATUS.ACTIVE)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
