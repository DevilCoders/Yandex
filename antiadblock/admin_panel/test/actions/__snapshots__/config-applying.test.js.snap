// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-applying actions should create START_ACTIVE_CONFIG_LOADING and END_ACTIVE_CONFIG_LOADING 1`] = `
Array [
  Object {
    "labelId": 123,
    "target": "active",
    "type": "START_ACTIVE_CONFIG_LOADING",
  },
  Object {
    "config": Object {
      "comment": "made a config",
      "data": Object {
        "field1": "value1",
        "field2": "value2",
        "field3": "value3",
      },
      "date": "2018-01-10T11:12:13",
      "id": 234,
      "statuses": Array [
        Object {
          "comment": "",
          "status": "active",
        },
      ],
    },
    "labelId": 123,
    "target": "active",
    "type": "END_ACTIVE_CONFIG_LOADING",
  },
]
`;

exports[`config-applying actions should create START_ACTIVE_CONFIG_LOADING and SET_ERRORS 1`] = `
Array [
  Object {
    "labelId": 123,
    "target": "active",
    "type": "START_ACTIVE_CONFIG_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`config-applying actions should create START_ACTIVE_CONFIG_LOADING and try to fetch ACTIVE config after error 1`] = `
Array [
  Object {
    "labelId": 123,
    "target": "test",
    "type": "START_ACTIVE_CONFIG_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`config-applying actions should create START_ACTIVE_CONFIG_LOADING and try to fetch ACTIVE config after error 2`] = `
Array [
  Object {
    "labelId": 123,
    "target": "active",
    "type": "START_ACTIVE_CONFIG_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;
